test/functional/Exception/CouldNotRenderExceptionTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Dhii\Output\FuncTest\Exception;
+
+use Xpmock\TestCase;
+use Exception as RootException;
+use Dhii\Output\RendererInterface;
+use Dhii\Output\Exception\CouldNotRenderException as TestSubject;
+
+/**
+ * Tests {@see TestSubject}.
+ *
+ * @since [*next-version*]
+ */
+class CouldNotRenderExceptionTest extends TestCase
+{
+    /**
+     * The name of the test subject.
+     *
+     * @since [*next-version*]
+     */
+    const TEST_SUBJECT_CLASSNAME = 'Dhii\Output\Exception\CouldNotRenderException';
+
+    /**
+     * Creates a new instance of the test subject.
+     *
+     * @since [*next-version*]
+     *
+     * @return TestSubject The new instance of the test subject.
+     */
+    public function createInstance($message = null, $code = null, $previous = null, $renderer = null)
+    {
+        $mock = $this->mock(static::TEST_SUBJECT_CLASSNAME)
+                ->new($message, $code, $previous, $renderer);
+
+        return $mock;
+    }
+
+    /**
+     * Tests whether a valid instance of the test subject can be created.
+     *
+     * @since [*next-version*]
+     */
+    public function testCanBeCreated()
+    {
+        $subject = $this->createInstance();
+
+        $this->assertInstanceOf(self::TEST_SUBJECT_CLASSNAME, $subject, 'A valid instance of the test subject could not be created');
+        $this->assertInstanceOf('Dhii\Output\Exception\RendererExceptionInterface', $subject, 'Subject does not implement required interface');
+        $this->assertInstanceOf('Dhii\Output\Exception\CouldNotRenderExceptionInterface', $subject, 'Subject does not implement required interface');
+    }
+
+    /**
+     * Creates a new exception.
+     *
+     * @since [*next-version*]
+     *
+     * @param string        $message
+     * @param int           $code
+     * @param RootException $previous
+     *
+     * @return RootException The new exception.
+     */
+    public function createException($message = null, $code = null, RootException $previous = null)
```

Comment:
Here and in every other test, I would prefer if the order of methods in test classes to be as follows, without any mixing:

* utility and factory methods first
* test methods second

Reply:
Agreed. Can we do this later? This is kind of a pressing thing, and implementing your requested changes already set me back an hour and a half.

Reply:
Sure. This is just for code tidiness more than a blocker.